### ISSUE TITLE
False positives with thecodingmachine/safe v3

### ISSUE THREAD

**Issue by mdellamina** [maglnet/ComposerRequireChecker#575]
After having updated [thecodingmachine/safe](https://github.com/thecodingmachine/safe) to the v3.0.2 composer-require-checker fails with the following error

```
The following 15 unknown symbols were found:
+-------------------------------+--------------------+
| Unknown Symbol                | Guessed Dependency |
+-------------------------------+--------------------+
| Safe\array_replace            |                    |
| Safe\base64_decode            |                    |
| Safe\class_uses               |                    |
| Safe\file_get_contents        |                    |
...more lines...
```

Looking at what changed in the v3 the files listed in autoload.files of the composer.json don't actually have the function declaration written inside but do a conditional require based on the version of php.

```php
// https://github.com/thecodingmachine/safe/blob/master/generated/filesystem.php
<?php

if (str_starts_with(PHP_VERSION, "8.1.")) {
    require_once __DIR__ . '/8.1/filesystem.php';
}
if (str_starts_with(PHP_VERSION, "8.2.")) {
    require_once __DIR__ . '/8.2/filesystem.php';
}
if (str_starts_with(PHP_VERSION, "8.3.")) {
    require_once __DIR__ . '/8.3/filesystem.php';
}
if (str_starts_with(PHP_VERSION, "8.4.")) {
    require_once __DIR__ . '/8.4/filesystem.php';
}
if (str_starts_with(PHP_VERSION, "8.5.")) {
    require_once __DIR__ . '/8.5/filesystem.php';
}
```

I guess those files are directly parsed for symbols and what not and are not actually executed so I tried to add those to the `scan-files` configuration but with no success.

```json
{
  "symbol-whitelist" : [
    "null", "true", "false", "static", "self", "parent", "array", "string", "int", "float", "bool", "iterable", "callable", "void", "object", "mixed"
  ],
  "php-core-extensions" : [
    "Core",  "date", "pcre", "Phar", "Reflection", "SPL", "random", "standard"
  ],
  "scan-files" : [
    "vendor/thecodingmachine/safe/generated/8.3/*.php",
    "bin/console",
    "bin/*.php",
    "public/*.php"
  ]
}

```
I also get the same error listing all individual files. I guess I'm doing something wrong but I can't figure out what.

**Comment by Ocramius** [comment]
> I tried to add those to the `scan-files` configuration but with no success.

Does that setting even support globbing? I'd suggest checking with individual listed files first.

I don't think dynamic includes will ever be supported, as we're moving even further towards statically analyzing code...

**Comment by mdellamina** [comment]
You are right, I managed to get it working listing only the files containing the functions we are using.

BTW: globbing does work but requiring everything will include function that need additional extensions/dependencies and as such generating even more errors.
